pkg/reconciler/buildrun/resources/conditions.go

```diff
@@ -95,29 +95,41 @@ func UpdateBuildRunUsingTaskRunCondition(ctx context.Context, client client.Clie
 			}
 
 			//nolint:staticcheck // SA1019 we want to give users some time to adopt to failureDetails
-			buildRun.Status.FailureDetails = &buildv1beta1.FailureDetails{
-				Location: &buildv1beta1.Location{
+			buildRun.Status.FailureDetails = &buildapi.FailureDetails{
+				Location: &buildapi.Location{
 					Pod: pod.Name,
 				},
 			}
 
 			if pod.Status.Reason == "Evicted" {
 				message = pod.Status.Message
-				reason = buildv1beta1.BuildRunStatePodEvicted
+				reason = buildapi.BuildRunStatePodEvicted
 				if failedContainer != nil {
 					//nolint:staticcheck // SA1019 we want to give users some time to adopt to failureDetails
 					buildRun.Status.FailureDetails.Location.Container = failedContainer.Name
 				}
 			} else if failedContainer != nil {
+
 				//nolint:staticcheck // SA1019 we want to give users some time to adopt to failureDetails
 				buildRun.Status.FailureDetails.Location.Container = failedContainer.Name
-				message = fmt.Sprintf("buildrun step %s failed in pod %s, for detailed information: kubectl --namespace %s logs %s --container=%s",
-					failedContainer.Name,
-					pod.Name,
-					pod.Namespace,
-					pod.Name,
-					failedContainer.Name,
-				)
+
+				if failedContainer.Name == "step-image-processing" && exitCode == 22 {
+					reason = buildapi.BuildRunStateVulnerabilitiesFound
+					message = fmt.Sprintf("Vulnerabilities have been found in the image. For detailed information,see kubectl --namespace %s logs %s --container=%s",
```

Comment:
Instead or in addition to pointing a user to the logs, could we also point to the status fields that most like are populated or am I confusing things here?